uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/leave-request-action-dropdown.html

```diff
@@ -0,0 +1,20 @@
+<div uib-dropdown dropdown-append-to-body ng-if="vm.actions.length">
+  <a href class="btn btn-default btn-inline" uib-dropdown-toggle>
+    <i class="fa fa-ellipsis-v"></i>
+  </a>
+  <ul class="dropdown-menu dropdown-menu-right" uib-dropdown-menu>
+    <li ng-repeat="action in vm.actions"
+      ng-switch="['edit', 'respond', 'view'].indexOf(action) > -1">
```

Comment:
What if you put in each "action" object a simple `openPopup` boolean property? So you could simply do
```html
<li ng-repeat="action in vm.actions" ng-switch="action.openPopup">
  <a href ng-switch-when="true" leave-request-popup>
  <a href ng-switch-when="false" ng-click="vm.act(action)">
```
the markup would be much cleaner and you would remove business logic from the view